### PR TITLE
Multiple Updates to WinRM Transport

### LINKED ISSUE #7: Unpacking files using the COM objects overwrites directories.
When uploading the windows cookbook via the chef-zero provisioner, the cookbooks directory is either not expanded or it is overwritten at some point.

### PR DESCRIPTION
**Pull request by usepowershell**
Fix unzipping from COM.  Closes #7
Fix failed single file copies (when parent directory was not present).
Adding more tests around decode_files.ps1
Cleaner error output from @xmik

### CODE REVIEW

**Comment by usepowershell** [comment]
@tyler-ball @fnichol These are some pretty necessary updates.  I broke COM uploading when I fixed IO.Compression last time.  I've got that sorted out and added a bunch of tests around the PowerShell to cover those cases.  I've also got the single file copy fail fixed and a test to cover that.

**Comment by tyler-ball** [comment]
This LGTM, but I could use some help groking the Powershell - @jdmundrawala

**Comment by jaym** [comment]
@smurawski this fixes the issue i was having
You are a god

File: lib/winrm/transport/file_transporter.rb [hunk 1]
```diff
@@ -252,8 +252,8 @@ def decode_files(files)
           vars = %{$hash_file = "#{hash_file}"\n}
 
           output = service.run_powershell_script(
-            [vars,
-             ". #{decode_files_script}",
+            ["#{decode_files_script}",
```
Comment (jaym) [inline lib/winrm/transport/file_transporter.rb:255]: unneeded string interpolation

Comment (jaym) [inline lib/winrm/transport/file_transporter.rb:255]: For my own learning, what does this change do

**Comment by jaym** [comment]
:100: please :ship:

**Comment by usepowershell** [comment]
@jdmundrawala moving the decode-files call out of the script file makes it easier to test the powershell. Then the script file is just a library of functions and we don't have to try to intercept the invocation to test the internal functions.

**Comment by tyler-ball** [comment]
We have 2 :+1: - I'm going to merge this unless someone stops me

**Comment by jaym** [comment]
merge

**Comment by fnichol** [comment]
I'm seeing pretty decisive :shipit:, so yay!

**Pull request merged**